### ISSUE TITLE
Appearance property "base" works incorrect, if appearance from parent theme has string value.

### ISSUE THREAD

**Issue by mfedyashev** [qooxdoo/qooxdoo#9429]
For example, appearance for "datefield" from theme "qx.theme.modern.Appearance" has value:

> "datefield" : "combobox",

If create custom theme and extend "qx.theme.modern.Appearance", then in custom theme create apearance with the same name "datefield" and use property "base: true":

`"datefield": {
      base: true,
      style: function (states) {
          return {
              textColor: "red"
          };
      }
 }`

In this case the result appearance does not have properties from parent theme.

Qooxdoo Playground example: http://tinyurl.com/y8hj6b82

**Comment by cajus** [comment]
Didn't you mean to change the textColor of *datefield/textfield*?

**Comment by mfedyashev** [comment]
No, I didn't mean textColor. You can use empty style function, for example:
`style: function (states) { return { }; }`
I mean, that datefield appearance in custom theme does not inherite properties from parent theme.

**Comment by cajus** [comment]
I can't find a problem in your description. What I ment with my comment was that there *is no textColor* on the level of *datefield* that is used somewhere, so it will not have any effect.

Maybe the problem is different: `base: true` does not work if the parent class has no own top level styling. In the modern theme, _datefield_ is just directly aliased to _combobox_. So you might need to add an _include: 'combobox'_ to your definition to make it work.

**Comment by mfedyashev** [comment]
Problem is not about textColor, I know that textColor on the level datefield does not have any effect. It just was bad example.
Issue is the Date field with custom theme does not have border like the Date field with moderh theme.
So, as User, I set "base: true" and expect that custom datefield will include all property from Modern datefield. But, actually, it does not work.

Yes, _include: 'combobox'_ can help in this case, but, I think, it just workaround.

**Comment by cajus** [comment]
It reacts like the comments in the appearance manager suggest. So that was done this way by intention of the original authors. I don't know why they did it this way.

**Comment by mfedyashev** [comment]
The function _qx.theme.manager.Appearance.styleFrom_ uses common **this.__aliasMap** for all themes. For my custom theme datefield appearance is Entry, so when _styleFrom_ calculates this appearnce, item _"datefield": "datefield"_ is added to alias map. Then _styleFrom_ calculates datefield appearance from Modern theme, but it does not resolve new id, it just gets existing alias from map. But alias _"datefield": "datefield"_ is not correct for Modern theme. For Modern theme it should be **"datefield": "combobox"**

**Comment by ronkie** [comment]
I've slightly modified an example:

https://tinyurl.com/yafwjr97

If we want to style text color of a textfield as subcontrol of a datefield, the issue is also reproducible.
You can notice that even though only text color is styled, datefield/textfield also loses it's paddings (they are not picked up anymore from the parent appearance).

Compare this with styling combobox/datefield (it is commented out in the example) - in this case paddings and other stylings are not lost, because they are still picked up from the parent theme.

**Comment by ronkie** [comment]
So, it looks like "base" call for an appearance that overrides appearance with alias set through "string form" (e.g. "datefield/textfield": "combobox/textfield") does not do it's job, because not all the style properties of parent appearance are picked up.

We can put in the parent theme, instead of string alias:

"datefield/textfield": {
            alias: "combobox/textfield",
            include: "combobox/textfield",
            style: function() {
                return {};
            }
        }

In this case everything will be mapped correctly, and also in inherited appearance all the styles will be picked up correctly.

The "aliases" section of docs:

http://www.qooxdoo.org/current/pages/desktop/ui_appearance.html#aliases

Says "There is no type of implicit inheritance." So does this mean in particular that properties that are obtained through alias will not be inherited?
